client/state/test/index.js

```diff
@@ -0,0 +1,46 @@
+require( 'lib/react-test-env-setup' )();
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import mockery from 'mockery';
+
+describe( 'state', () => {
+	var createReduxStore;
+	before( () => {
+		mockery.registerMock( 'lib/themes/middlewares.js', {
+			// stub redux middleware
+			// see: http://rackt.org/redux/docs/advanced/Middleware.html
+			analyticsMiddleware: store => next => action => { return next( action ); }
+		} );
+		mockery.enable( {
+			warnOnReplace: false,
+			warnOnUnregistered: false
+		} );
+		createReduxStore = require( 'state' ).createReduxStore;
+	} );
+	after( () => {
+		mockery.deregisterAll();
+		mockery.disable();
+	} );
+	describe( 'createReduxStore', () => {
+		it( 'can be called without specifying initialState', () => {
+			const reduxStoreNoArgs = createReduxStore().getState();
+			const reduxStoreWithEmptyState = createReduxStore( {} ).getState();
+			const reduxStoreWithNullState = createReduxStore( null ).getState();
+			expect( reduxStoreNoArgs ).to.eql( reduxStoreWithEmptyState );
+			expect( reduxStoreWithNullState ).to.eql( reduxStoreWithEmptyState );
+		} );
```

Comment:
The `createReduxStore( state ).getState()` looks a bit like this. I'm going to add a simple check that it's an object instead. A fixture in the root directory would be pretty fragile, but I think we should have something like this on the module level.
```
{
	"plugins": {
		"wporg": {}
	},
	"notices": {
		"items": []
	},
	"posts": {
		"items": {},
		"sitePosts": {},
		"siteQueries": {},
		"siteQueriesLastPage": {}
	},
	"sharing": {
		"publicize": {
			"fetchingConnections": {},
			"connections": {},
			"connectionsBySiteId": {}
		}
	},
	"sites": {
		"items": {},
		"plans": {}
	},
	"siteSettings": {
		"exporter": {
			"ui": {
				"exportingState": "READY",
				"postType": null
			}
		}
	},
	"themes": {
		"themes": {
			"themes": {},
			"currentSiteId": 0
		},
		"themesList": {
			"list": [],
			"nextId": 1,
			"query": {
				"search": "",
				"perPage": 20,
				"page": 0,
				"id": 0
			},
			"queryState": {
				"isLastPage": false,
				"isFetchingNextPage": false
			},
			"active": 0
		},
		"themesLastQuery": {
			"previousSiteId": 0,
			"currentSiteId": null,
			"isJetpack": null,
			"lastParams": null
		},
		"currentTheme": {
			"isActivating": false,
			"hasActivated": false,
			"currentThemes": {}
		}
	},
	"users": {
		"items": {}
	},
	"currentUser": {
		"id": null
	},
	"ui": {
		"section": false,
		"isLoading": false,
		"hasSidebar": true,
		"chunkName": false,
		"selectedSiteId": null,
		"editor": {
			"media": {
				"advanced": false
			}
		}
	}
}
```